pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix

```diff
@@ -0,0 +1,144 @@
+{ lib
+, fetchurl
+, runCommand
+
+  # script interpteers
+, bash
+, jdk
+, perl
+, python3
+, ruby
+, snobol4
+, tk
+
+  # compiled binaries
+, texliveBinaries
+  # fixed hashes
+, texliveFixedHashes
+  # mirrors
+, texliveMirrors
+}:
+
+pname:
+
+{ revision
+, version ? toString revision
+, sha512
```

Comment:
Why not a generic hash?

Reply:
It's guaranteed to be sha512 because it comes from `tlpdb.nix`. We can leave it as it is while `buildTeXLivePackage` is private.